runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/util/StringValuesMap.kt

```diff
@@ -81,6 +81,22 @@ internal open class StringValuesMapImpl(
     override fun contains(name: String, value: String): Boolean = getAll(name)?.contains(value) ?: false
 
     override fun isEmpty(): Boolean = values.isEmpty()
+
+    override fun equals(other: Any?): Boolean =
+        other is StringValuesMap && names().size == other.names().size &&
+            names().all {
```

Comment:
I think this can be simplified to:

```kotlin
other is StringValuesMap && 
     names().size == other.names().size && 
     names().all { getAll(it) == other.getAll(it) }
```

if it's a case insensitive map then the [get implementation](https://github.com/awslabs/smithy-kotlin/blob/8d17f0666d9bec313d60838e8e00bdef950f7a13/runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/util/CaseInsensitiveMap.kt#L33) takes care of that